libexec/rtld-elf/rtld_c18n.c

```diff
@@ -826,7 +834,16 @@ resolve_untrusted_stk_impl(stk_table_index index)
 	void *stk;
 	sigset_t nset, oset;
 	struct stk_table *table;
-	struct trusted_frame backup, *tf;
+	struct trusted_frame *tf = get_trusted_stk();
+
+	/*
+	 * Push a dummy trusted frame indicating that the current compartment is
+	 * RTLD.
+	 */
+	*--tf = (struct trusted_frame) {
```

Comment:
This seems to be mixing in stuff from https://github.com/CTSRD-CHERI/cheribsd/pull/2134 if I'm not mistaken? If so, updating it to reflect @jrtc27's comments from there might make sense? This is more of a general comment for the rtld changes rather than this specific spot.

Reply:
Yeah I rebased on #2134 to see if everything works together as a whole. This PR remains focused on the unwind-related bits.